### PR TITLE
Force new private users to accept the privacy policy before creating an account

### PR DESCRIPTION
**Pull request by jorg-vr**
This pull request forces new private users to accept the privacy policy before creating an account. Replaces #3803 
Institutional users can still sign up without prompt.

image
image

- [x] Tests were added

Part of https://github.com/dodona-edu/dodona/issues/2597 .

### CODE REVIEW

File: config/locales/views/auth/nl.yml [hunk 1]
```diff
@@ -22,3 +22,7 @@ nl:
       providers_title: "Accounts koppelen"
       "contact_support_html": "Ben jij dit niet of blijf je problemen hebben? Vul het %{form} in en we helpen je zo snel mogelijk verder."
       contact_form: "contactformulier"
+    privacy_prompt:
+      text_html: "Om Dodona te gebruiken moet je onze privacyverklaring accepteren. Op de <a href=\"%{your_data}\" target=\"_blank\">jouw data</a> pagina leggen we in mensentaal uit welke data we verzamelen en hoe we die gebruiken. De juridisch bindende versie kan je in onze <a href=\"%{privacy_statement}\" target=\"_blank\">privacyverklaring</a> vinden."
+      accept_button: Accepteer de privacyverklaring
```
Comment (niknetniko) [inline config/locales/views/auth/nl.yml:27]:
```suggestion
      accept_button: Privacyverklaring accepteren
```

File: config/locales/views/auth/nl.yml [hunk 2]
```diff
@@ -22,3 +22,7 @@ nl:
       providers_title: "Accounts koppelen"
       "contact_support_html": "Ben jij dit niet of blijf je problemen hebben? Vul het %{form} in en we helpen je zo snel mogelijk verder."
       contact_form: "contactformulier"
+    privacy_prompt:
+      text_html: "Om Dodona te gebruiken moet je onze privacyverklaring accepteren. Op de <a href=\"%{your_data}\" target=\"_blank\">jouw data</a> pagina leggen we in mensentaal uit welke data we verzamelen en hoe we die gebruiken. De juridisch bindende versie kan je in onze <a href=\"%{privacy_statement}\" target=\"_blank\">privacyverklaring</a> vinden."
```
Comment (niknetniko) [inline config/locales/views/auth/nl.yml:26]:
```suggestion
      text_html: "Om Dodona te gebruiken moet je onze privacyverklaring accepteren. Op de pagina <a href=\"%{your_data}\" target=\"_blank\">jouw data</a> leggen we in mensentaal uit welke data we verzamelen en hoe we die gebruiken. De juridisch bindende versie kan je in onze <a href=\"%{privacy_statement}\" target=\"_blank\">privacyverklaring</a> vinden."
```

**Review by niknetniko — CHANGES_REQUESTED** [review]

**Review by niknetniko — APPROVED** [review]

**Review by chvp — COMMENTED** [review]
Looks good, but if I understood correctly this will be changed to not force an accept for institutional users?

**Comment by jorg-vr** [comment]
@chvp yes I will update this, so that the new behavior only applies for personal accounts

**Review by bmesuere — APPROVED** [review]

**Review by chvp — APPROVED** [review]

**Pull request merged**